#1: Importing Libraries and Loading Data

In [1]:
import pandas as pd

# Load the dataset
df = pd.read_csv('Day7_Student_Dataset.csv')

# Inspect the first 5 rows
print("--- First 5 Rows ---")
display(df.head())

# Inspect the last 5 rows
print("\n--- Last 5 Rows ---")
display(df.tail())

# View a random sample of 5 rows
print("\n--- Random Sample of 5 Rows ---")
display(df.sample(5))

--- First 5 Rows ---


,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
0,1001,Aarav Sharma,19,Data Science,87.0,91.0,Srinagar
1,1002,Zoya Khan,20,AI,92.0,95.0,Delhi
2,1003,Rohan Mehta,18,Computer Science,76.0,82.0,Mumbai
3,1004,Ananya Singh,21,Data Science,81.0,88.0,Srinagar
4,1005,Kabir Ali,19,AI,69.0,76.0,Lucknow



--- Last 5 Rows ---


,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
20,1021,Aman Kumar,18,Computer Science,74.0,78.0,Delhi
21,1022,Alina Shah,19,Data Science,96.0,98.0,Srinagar
22,1023,Nikhil Sood,22,AI,78.0,83.0,Pune
23,1024,Maryam Khan,20,Computer Science,85.0,88.0,Hyderabad
24,1025,Reyansh Jain,21,Data Science,80.0,84.0,Kolkata



--- Random Sample of 5 Rows ---


,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
10,1011,Vivaan Kapoor,20,AI,91.0,94.0,Jaipur
14,1015,Karan Joshi,20,Computer Science,77.0,85.0,Chandigarh
7,1008,Sara Ahmed,19,AI,73.0,NaN,Hyderabad
1,1002,Zoya Khan,20,AI,92.0,95.0,Delhi
2,1003,Rohan Mehta,18,Computer Science,76.0,82.0,Mumbai


#2: Inspecting Dataset Structure

In [2]:
# Check the shape of the dataset (rows, columns)
print("--- Dataset Shape ---")
print(df.shape)

# View the column names
print("\n--- Columns ---")
print(df.columns.tolist())

# View the index details
print("\n--- Index ---")
print(df.index)

# Check the data types of each column
print("\n--- Data Types ---")
print(df.dtypes)

--- Dataset Shape ---
(25, 7)

--- Columns ---
['Student_ID', 'Name', 'Age', 'Course', 'Marks', 'Attendance_Percentage', 'City']

--- Index ---
RangeIndex(start=0, stop=25, step=1)

--- Data Types ---
Student_ID                 int64
Name                      object
Age                        int64
Course                    object
Marks                    float64
Attendance_Percentage    float64
City                      object
dtype: object


#3: Summary Statistics and Missing Values

In [3]:
# Generate a concise summary of the DataFrame
print("--- Dataset Info ---")
df.info()

# Generate descriptive statistics for all columns
print("\n--- Descriptive Statistics ---")
display(df.describe(include='all'))

# Check for missing values in each column
print("\n--- Missing Values Count ---")
print(df.isnull().sum())

--- Dataset Info ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25 entries, 0 to 24
Data columns (total 7 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Student_ID             25 non-null     int64  
 1   Name                   25 non-null     object 
 2   Age                    25 non-null     int64  
 3   Course                 25 non-null     object 
 4   Marks                  24 non-null     float64
 5   Attendance_Percentage  24 non-null     float64
 6   City                   24 non-null     object 
dtypes: float64(2), int64(2), object(3)
memory usage: 1.5+ KB

--- Descriptive Statistics ---


,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
count,25.000000,25,25.000000,25,24.000000,24.000000,24
unique,NaN,25,NaN,3,NaN,NaN,13
top,NaN,Aarav Sharma,NaN,Data Science,NaN,NaN,Srinagar
freq,NaN,1,NaN,9,NaN,NaN,6
mean,1013.000000,NaN,20.000000,NaN,82.250000,87.083333,NaN
std,7.359801,NaN,1.414214,NaN,8.689074,7.252686,NaN
min,1001.000000,NaN,18.000000,NaN,68.000000,72.000000,NaN
25%,1007.000000,NaN,19.000000,NaN,75.500000,82.750000,NaN
50%,1013.000000,NaN,20.000000,NaN,82.500000,88.000000,NaN
75%,1019.000000,NaN,21.000000,NaN,89.250000,92.250000,NaN



--- Missing Values Count ---
Student_ID               0
Name                     0
Age                      0
Course                   0
Marks                    1
Attendance_Percentage    1
City                     1
dtype: int64


### **Dataset Observations & Summary**

Based on the execution of the code above, here are the core observations regarding the `Day7_Student_Dataset.csv` dataset:

*   **Size of Dataset:** The dataset contains 25 rows (student records) and 7 columns (features).
*   **Available Features:** The dataset captures the following columns: `Student_ID`, `Name`, `Age`, `Course`, `Marks`, `Attendance_Percentage`, and `City`.
*   **Data Types:**
    *   **Numeric (Integer):** `Student_ID`, `Age` (`int64`)
    *   **Numeric (Float):** `Marks`, `Attendance_Percentage` (`float64`)
    *   **Categorical/Text:** `Name`, `Course`, `City` (`object`)
*   **Missing Values:** The dataset is mostly complete, but there are slightly incomplete records. There is exactly 1 missing value in the `Marks` column, 1 missing value in the `Attendance_Percentage` column, and 1 missing value in the `City` column.
*   **Descriptive Statistics Highlights:**
    *   **Age:** The students' ages range from 18 to 23 years, with the average age being exactly 20.
    *   **Marks & Attendance:** Out of the 24 recorded scores, the average mark is 82.25, with the highest being 96. Attendance averages 87.08%.
    *   **Demographics & Courses:** There are 3 unique courses represented, with Data Science being the most popular (9 students). Srinagar is the most frequently appearing city in the dataset.